sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -157,39 +156,74 @@ case class CreateDataSourceTableAsSelectCommand(
           // Since the table already exists and the save mode is Ignore, we will just return.
           return Seq.empty[Row]
         case SaveMode.Append =>
+          val existingTable = sessionState.catalog.getTableMetadata(tableIdentWithDB)
+          if (existingTable.tableType == CatalogTableType.VIEW) {
+            throw new AnalysisException("Saving data into a view is not allowed.")
```

Comment:
We already have [an assert](https://github.com/cloud-fan/spark/blob/b1dbd0a19a174eaae1aaf114e04f6d3683ea65c4/sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala#L133) at the beginning of this function. We need to improve the error message there. That can cover more cases.

So far, the error message is pretty confusing.
```
assertion failed
java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:156)
```

We can add a test case for it.
```Scala
        val df = spark.range(1, 10).toDF("id1")
        df.write.saveAsTable("tab1")
        spark.sql("create view view1 as select * from tab1")
        df.write.mode(SaveMode.Append).format("parquet").saveAsTable("view1")
```